src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1856,6 +1859,31 @@ impl Coordinator {
         Ok(ExecuteResponse::SetVariable { name, reset: true })
     }
 
+    pub(super) fn sequence_set_transaction(
+        &self,
+        session: &mut Session,
+        plan: SetTransactionPlan,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        // TODO(jkosh44) Only supports isolation levels for now.
+        for mode in plan.modes {
+            match mode {
+                TransactionMode::AccessMode(_) => {
+                    return Err(AdapterError::Unsupported("SET TRANSACTION <access-mode>"))
+                }
+                TransactionMode::IsolationLevel(isolation_level) => session.vars_mut().set(
+                    Some(self.catalog().system_config()),
```

Comment:
What does it mean for a transaction to change itself from serializable to strict mid transaction? Does the addition of linearizability mean that the transaction is now possibly invalid? I'm not super concerned about this, more wondering.

Reply:
https://github.com/MaterializeInc/materialize/pull/19724